docs/source/reference/dataset_formats.rst

```diff
@@ -29,8 +29,215 @@ Recommendations
+dimension types.
+
+To rename columns, add the ``dimension_type`` field to the column definition. This tells dsgrid
+what dimension the column represents, and dsgrid will automatically rename it at runtime.
+
+This feature works for all file formats (Parquet, CSV, JSON), not just CSV files.
+
+Example with custom column names:
+
+.. code-block:: JavaScript
+
+    data_file: {
+      path: "load_data.parquet",
+      columns: [
+        {
+          name: "county",           // Actual column name in the file
+          dimension_type: "geography",  // Will be renamed to "geography"
+        },
+        {
+          name: "end_use",          // Actual column name in the file
+          dimension_type: "metric",     // Will be renamed to "metric"
+        },
+        {
+          name: "building_type",    // Actual column name in the file
+          dimension_type: "subsector",  // Will be renamed to "subsector"
+        },
+      ],
+    }
+
+You can combine ``dimension_type`` with ``data_type`` when using CSV files:
+
+.. code-block:: JavaScript
+
+    data_file: {
+      path: "load_data.csv",
+      columns: [
+        {
+          name: "fips_code",
+          data_type: "STRING",
+          dimension_type: "geography",
+        },
+        {
+          name: "fuel_type",
+          data_type: "STRING",
+          dimension_type: "metric",
+        },
+        {
+          name: "consumption",
+          data_type: "DOUBLE",
+          // No dimension_type - column name stays as "consumption"
+        },
+      ],
+    }
+
+
+Ignoring Columns
+================
+Data files may contain columns that are not needed for dsgrid processing. Rather than
+modifying your source files, you can tell dsgrid to ignore (drop) specific columns when
+reading the data.
+
+To ignore columns, add an ``ignore_columns`` field to the ``data_file`` (or ``lookup_data_file``)
+section in your dataset configuration. This field accepts a list of column names to drop.
+
+This feature works for all file formats (Parquet, CSV, JSON).
+
+Example with ignored columns:
+
+.. code-block:: JavaScript
+
+    data_file: {
+      path: "load_data.parquet",
+      ignore_columns: ["internal_id", "notes"],
+    }
+
+You can combine ``ignore_columns`` with ``columns`` for type overrides and renaming:
+
+.. code-block:: JavaScript
+
+    data_file: {
+      path: "load_data.csv",
+      columns: [
+        {
+          name: "county",
+          data_type: "STRING",
+          dimension_type: "geography",
+        },
+        {
+          name: "value",
+          data_type: "DOUBLE",
+        },
+      ],
+      ignore_columns: ["intenal_id", "notes"],
+    }
+
+Note that a column cannot appear in both ``columns`` and ``ignore_columns`` - dsgrid will
+raise an error if there is any overlap.
+
+Columns are dropped after the file is read but before any column renaming occurs. This means
+you should use the original column names from the file in the ``ignore_columns`` list.
```

Comment:
I'm not sure this is needed. We aren't offering to rename the ignored columns (which can't overlap with the columns that can be renamed).